extensions/vscode/src/extension/parseUsesSlug.ts

```diff
@@ -0,0 +1,53 @@
+const USES_PATTERN_STRING = "\\buses:\\s*(.+)$";
+
+export function parseUsesSlug(
+  lineText: string,
+  position?: number,
+): string | undefined {
+  const [slug, index] = findSlug(lineText, position);
+  if (slug) {
+    return slug;
+  }
+  return undefined;
+}
+
+export function findSlug(
+  lineText: string,
+  position?: number,
+): [string | undefined, number | undefined] {
+  const pattern = new RegExp(USES_PATTERN_STRING, "dg");
+  let match;
+  while ((match = pattern.exec(lineText)) !== null) {
+    let slug = match[1].trim();
+    let quoteOffset = 0;
+
+    // Check for surrounding quotes
+    const quoteMatch = slug.match(/^(['"])(.*)\1/);
+    if (quoteMatch) {
+      // If quoted, remove the quotes but keep everything inside (including #)
+      slug = quoteMatch[2].trim();
+      quoteOffset = 1; // Add offset to skip the opening quote
+    } else {
+      // If not quoted, remove any trailing comment
+      slug = slug.replace(/\s*#.*$/, "").trim();
+    }
+
+    if (!slug || !slug.startsWith("$")) {
+      continue;
+    }
+
+    const indices = match.indices![1];
+    const startPosition = indices[0] + quoteOffset;
+    const endPosition = indices[1] + quoteOffset;
```

Comment:
This considers the trailing part of the line after the close quote part of the slug right?

Reply:
yeah that's wrong, end quote should not be included. Should be endPosition = indices[1] -  quoteOffset. I added another test to check that.